autoload/codefmt/clangformat.vim

```diff
@@ -182,8 +182,8 @@ function! codefmt#clangformat#GetFormatter() abort
         let [l:new_line, l:new_col] = s:OffsetToPosition(l:offset, l:formatted)
         call cursor(l:new_line, l:new_col)
       catch
-        call maktaba#error#Warn('Unable to parse clang-format cursor pos: %s',
-            \ v:exception)
+        call maktaba#error#Warn(v:exception)
```

Comment:
What's the motivation for changing the warning? Ideally the context is printed before the exception message. Maybe separating with a newline instead of a space would help with what you're going for?

Reply:
Oh, I believe that's more a convention for error messages *thrown* from library functions and actually used sort of accidentally with "Warn" there, where Warn is usually human-readable informational messages directed at users.

It sounds like the "regexing for unexpected errors" you're talking about is the behavior from the default ["GUESS-ERRORS" option](https://github.com/google/vroom/blob/abd0e4ad98254e7c245c58ce22bd1635179494f0/vroom/args.py#L237) for message strictness. That only has any effect for messages you don't otherwise match as expected (see [example in messages.vroom](https://github.com/google/vroom/blob/abd0e4ad98254e7c245c58ce22bd1635179494f0/examples/messages.vroom#L14)). You should be able to control that however you need without modifying the warning logging, but can you clarify what you're aiming to get the test to do?

Reply:
> can you clarify what you're aiming to get the test to do?

I'm trying to write a test that duplicates the bug.

The main approach I'm using is

1. Call `FormatCode`
2. intercept the system call and return a blank string
3. Make sure there aren't any errors

As it stands, since the code emits a warning, I don't have a very robust way of ensuring that `FormatCode` runs without any problems.

I _could_ check for the specific warning output, but even this doesn't help since the warning is only displayed when the error occurs. Once the error is fixed, there is no output, and I was unable to find a way to check for "the function call returns no output"

I can try the `--message=strict` thing, I must have missed that the handful of times I read through that document

Reply:
Oh, gotcha! You want to temporarily change message strictness to `STRICT` with the `@messages` directive ([example](https://github.com/google/vroom/blob/abd0e4ad98254e7c245c58ce22bd1635179494f0/examples/directives.vroom#L28-L45)).

Reply:
That's exactly what I needed :-D

Thanks for the help. Marking this as resolved now, though let me know if you have any other comments.